### PR TITLE
Mythicmobs 5.0.3 support & Minecraft 1.18.2 support

### PR DESCRIPTION
**Pull request by Axolottl**
Towny Resources wasn't working for 1.18.2 and Mythic Mobs 5.0.3

- Updated API for Mythic Mobs and its syntax
- Bumped up minecraft version to 1.18.2

### CODE REVIEW

File: src/main/java/io/github/townyadvanced/townyresources/controllers/TownResourceCollectionController.java [hunk 1]
```diff
@@ -100,8 +102,7 @@ private static List<ItemStack> buildItemStackList(Player player, Set<Entry<Strin
 
             // mythicmobs integration
             if (TownyResources.getPlugin().isMythicMobsInstalled()) {
-                ItemManager mythicItemManager = TownyResources.getPlugin().getMythicItemManager();
-                ItemStack mythicItem = mythicItemManager.getItemStack(materialName);
+                ItemStack mythicItem = MythicBukkit.inst().getItemManager().getItemStack(materialName);
```
Comment (LlmDl) [inline src/main/java/io/github/townyadvanced/townyresources/controllers/TownResourceCollectionController.java:105]:
```suggestion
                ItemStack mythicItem = TownyResources.getPlugin().getMythicItemManager().getItemStack(materialName);
```
Can probably do this.

File: src/main/java/io/github/townyadvanced/townyresources/settings/TownyResourcesSettings.java [hunk 2]
```diff
@@ -204,7 +206,8 @@ private static boolean isValidMaterial(String materialName) {
 		}
 		// mythicmobs integration
 		if (TownyResources.getPlugin().isMythicMobsInstalled()) {
-			ItemStack mythicItem = TownyResources.getPlugin().getMythicItemManager().getItemStack(materialName);
+			ItemStack mythicItem = MythicBukkit.inst().getItemManager().getItemStack(materialName);
```
Comment (LlmDl) [inline src/main/java/io/github/townyadvanced/townyresources/settings/TownyResourcesSettings.java:209]:
```suggestion
			ItemStack mythicItem = TownyResources.getPlugin().getMythicItemManager().getItemStack(materialName);
```

**Comment by Goosius1** [comment]
@Axolottl hi, what is the status of this PR?

**Comment by Axolottl** [comment]
Hey, I am closing it and making a new one when I fully fix up the commit

**Pull request closed without merging**